lib/components/map/connected-stops-overlay.js

```diff
@@ -21,25 +21,29 @@ const mapStateToProps = (state) => {
     patterns
   ) {
     // Display stops for a pattern or all patterns for a route.
-    let viewedPattern = patternId
 
-    // If a flex route is being shown but the pattern viewer is not active, then the
-    // stops of the first pattern of the route should be shown
-    // This will ensure that the flex zone stops are shown.
-
-    // Preferably, the flex stops would be rendered in a separate layer.
+    // Preferably, the flex zones would be rendered in a separate layer.
     // However, without changes to GraphQL, getting this data is very expensive
-    if (!viewedPattern && v2) {
-      viewedPattern = Object.keys(patterns)?.[0]
+    if (v2) {
+      // If a flex route is being shown, show flex zones from all patterns
+      // using the stops overlay, whether the pattern viewer is active or not.
+      Object.values(patterns).forEach((p) => {
+        stops = stops.concat(
+          p?.stops?.filter((s) => s.geometries?.geoJson?.type === 'Polygon')
+        )
+      })
     }
 
-    // Discard duplicates from pattern stops.
-    stops = (patterns?.[viewedPattern]?.stops || []).reduce((prev, cur) => {
-      if (!prev.find((stop) => stop.id === cur.id)) {
-        prev.push(cur)
-      }
-      return prev
-    }, [])
+    // Discard duplicates from pattern stops and flex zones.
+    stops = stops
+      .concat(patterns?.[patternId]?.stops || [])
```

Comment:
What does this `concat` line do?

Reply:
It was supposed to include flex stops in the duplicate stop removal, but good point, the two things are independent so concatenating was not necessary. The new logic for removing duplicate stops is in https://github.com/opentripplanner/otp-react-redux/pull/728/commits/f3b720e8128b7dcbd2a3716ab26e29153f4597da.